docs/service-fabric/refactor-migrated-app.md

```diff
@@ -0,0 +1,155 @@
+
+Tailspin sees other benefits in moving the Surveys application to a more granular architecture:
+* Each service can be packaged into independent projects with a scope small enough to be managed by a small team.
+* Each service can be independently versioned and deployed.
+* Each service can be implemented using the best technology for that service.
+
+For example, a service fabric cluster can include services built using different versions of the .Net Frameworks, Java, or other languages such as C or C++.
+
+> [!NOTE] 
+> Multitenancy is out of scope for the refactoring of this application. Tailspin has several options to support multitenancy and can make these design decisions later without affecting the initial design. For example, Tailspin can create separate instances of the services for each tenant within a cluster or create a separate cluster for each tenant.
+
+## Design considerations
+
+The following diagram shows the architecture of the Surveys application refactored to a more granular architecture:
+
+![](./images/surveys_03.png)
+
+**Tailspin.Web** is a stateless service self-hosting an ASP.NET MVC application that Tailspin customers visit to create surveys and view survey results. This service shares most of its code with the *Tailspin.Web* service from the ported Service Fabric application. As mentioned earlier, this service uses ASP.NET core and switches from using Kestrel as web frontend to implementing a WebListener.
+
+**Tailspin.Web.Surveys.Public** is a stateless service also self-hosting an ASP.NET MVC site. Users visit this site to select surveys from a list and then fill them out. This service shares most of its code with the *Tailspin.Web.Survey.Public* service from the ported Service Fabric application. This service also uses ASP.NET Core and also switches from using Kestrel as web frontend to implementing a WebListener.
+
+**Tailspin.SurveyResponseService** is a stateful service that stores survey answers in Azure Blob Storage. It also merges answers into the survey analysis data. The service is implemented as a stateful service because it uses a reliable queue to process survey answers in batches.
+
+**Tailspin.SurveyManagementService** is a stateless service that stores and retrieves surveys and survey questions. The service uses Azure Blob storage.
+
+**Tailspin.SurveyAnswerService** is a stateless service that retrieves survey answers and survey analysis. The service also uses Azure Blob storage.
+
+**Tailspin.SurveyAnalysisService** is a stateless service that persists survey answer summary data in a Redis cache. This service is called by the *Tailspin.SurveyResponseService* each time a survey is answered and the new survey answer data is merged in the summary data. In the original Cloud Services application, this functionality was implemented in the *Tailspin.Workers.Survey* role.
+
```

Comment:
Maybe add a short description of how the functionality was broken out into these services. Looks like the new services are SurveyResponse, SurveyManagement, and SurveyAnswer. So is it basically splitting domain logic out from the UI pieces? What are the benefits? Looks like the app is more complicated now. Or why not put all of the domain logic into a single middle tier?

Reply:
I've added a sentence that discusses the motivation for refactoring functionality into each service.